libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1047,15 +1045,31 @@ string YulUtilFunctions::storageArrayPushZeroFunction(ArrayType const& _type)
 				if iszero(lt(oldLen, <maxArrayLength>)) { <panic>() }
 				sstore(array, add(oldLen, 1))
 				slot, offset := <indexAccess>(array, oldLen)
-				<storeValue>(slot, offset, <zeroValueFunction>())
+				<?isValueType>
+					<storeValue>(slot, offset, <zeroValueFunction>())
+				<!isValueType>
+					if gt(offset, 0) { invalid() }
+					let endData := add(slot, <storageSize>)
+					for { let currSlot := slot } lt(currSlot, endData) { currSlot := add(currSlot, 1) } {
+						sstore(currSlot, 0)
+					}
```

Comment:
For @ekpyron's reference 
Here, we do not need to distinguish cases when member is of reference type, as we just set it's length to zero. In routine for deleting structs, it was needed as we needed to clear arrays properly.